pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1351,56 +1408,145 @@ func (vs *VSphere) NodeExists(nodeName k8stypes.NodeName) (bool, error) {
 	return false, nil
 }
 
-func (vs *VSphere) createDummyVM(ctx context.Context, datacenter *object.Datacenter, datastore *object.Datastore) (*object.VirtualMachine, error) {
+// A background routine which will be responsible for deleting stale dummy VM's.
+func (vs *VSphere) cleanUpDummyVMs(dummyVMPrefix string) {
+	// Create context
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	for {
+		// Ensure client is logged in and session is valid
+		vSphereLogin(ctx, vs)
```

Comment:
If you see the vSphereLogin(), if there an already existing valid session it would use it. Otherwise, it would create a new session. 

Since this is a background routine, the session be valid initially and might get expired after some time and we need to renew it.